checks.d/mysql.py

```diff
@@ -141,96 +355,193 @@ def _connect(self, host, port, mysql_sock, user, password, defaults_file):
+            results[
+                'Binlog_space_usage_bytes'] = self._get_binary_log_stats(db)
 
         # Compute key cache utilization metric
-        key_blocks_unused = self._collect_scalar('Key_blocks_unused', status_results)
-        key_cache_block_size = self._collect_scalar('key_cache_block_size', variables_results)
-        key_buffer_size = self._collect_scalar('key_buffer_size', variables_results)
-        key_cache_utilization = 1 - ((key_blocks_unused * key_cache_block_size) / key_buffer_size)
-        self.gauge("mysql.performance.key_cache_utilization", key_cache_utilization, tags=tags)
-
-        # Compute InnoDB buffer metrics
-        # Be sure InnoDB is enabled
-        if 'Innodb_page_size' in status_results:
-            page_size = self._collect_scalar('Innodb_page_size', status_results)
-            innodb_buffer_pool_pages_total = self._collect_scalar('Innodb_buffer_pool_pages_total',
-                                                                  status_results)
-            innodb_buffer_pool_pages_free = self._collect_scalar('Innodb_buffer_pool_pages_free',
-                                                                 status_results)
-            innodb_buffer_pool_pages_total = innodb_buffer_pool_pages_total * page_size
-            innodb_buffer_pool_pages_free = innodb_buffer_pool_pages_free * page_size
-            innodb_buffer_pool_pages_used = \
-                innodb_buffer_pool_pages_total - innodb_buffer_pool_pages_free
-
-            self.gauge("mysql.innodb.buffer_pool_free", innodb_buffer_pool_pages_free, tags=tags)
-            self.gauge("mysql.innodb.buffer_pool_used", innodb_buffer_pool_pages_used, tags=tags)
-            self.gauge("mysql.innodb.buffer_pool_total", innodb_buffer_pool_pages_total, tags=tags)
-
-            if innodb_buffer_pool_pages_total != 0:
-                innodb_buffer_pool_pages_utilization = \
-                    innodb_buffer_pool_pages_used / innodb_buffer_pool_pages_total
-                self.gauge("mysql.innodb.buffer_pool_utilization",
-                           innodb_buffer_pool_pages_utilization, tags=tags)
+        key_blocks_unused = self._collect_scalar('Key_blocks_unused', results)
+        key_cache_block_size = self._collect_scalar(
+            'key_cache_block_size', results)
+        key_buffer_size = self._collect_scalar('key_buffer_size', results)
+        key_cache_utilization = 1 - \
+            ((key_blocks_unused * key_cache_block_size) / key_buffer_size)
+
+        results['Key_buffer_size'] = key_buffer_size
+        results['Key_buffer_bytes_used'] = self._collect_scalar(
+            'Key_blocks_used', results) * key_cache_block_size
+        results['Key_buffer_bytes_unflushed'] = self._collect_scalar(
+            'Key_blocks_not_flushed', results) * key_cache_block_size
+        results['Key_cache_utilization'] = key_cache_utilization
+
+        VARS.update(VARIABLES_VARS)
```

Comment:
nitpick but usually UPPER_CASE variables are for constants. So you shouldn't update them.